### PR TITLE
Fix Whisper CI

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

### CODE REVIEW

File: src/transformers/generation/utils.py [hunk 1]
```diff
@@ -1452,10 +1452,11 @@ def _prepare_generated_length(
         ):
             generation_config.max_length -= inputs_tensor.shape[1]
         elif has_default_max_length:  # by default let's always generate 20 new tokens
-            generation_config.max_length = generation_config.max_length + input_ids_length
-            max_position_embeddings = getattr(self.config, "max_position_embeddings", None)
-            if max_position_embeddings is not None:
-                generation_config.max_length = min(generation_config.max_length, max_position_embeddings)
+            if generation_config.max_length == GenerationConfig().max_length:
```
Comment (ydshieh) [inline src/transformers/generation/utils.py:1455]: if `generation_config.max_length` is not the same as the default value (`20`), let's not change it!!!!

Comment (ArthurZucker) [reply src/transformers/generation/utils.py:1455]: should we not change `has_default_max_length` directly? 🤗

**Review by ArthurZucker — APPROVED** [review]
Thanks, this fixes quite some tests no? Can we make sure with a tiny pipeline that by default we generate more than 20 tokens?

**Comment by ydshieh** [comment]
> Thanks, this fixes quite some tests no? Can we make sure with a tiny pipeline that by default we generate more than 20 tokens?

Do you mean a new test case that will `generate more than 20 tokens`? Any motivation for such a new test?

File: src/transformers/generation/utils.py (hunk 1 above)
Comment (ydshieh) [reply src/transformers/generation/utils.py:1455]: I would rather avoid changing its original definition in this PR as it is also used in several places.

**Comment by ydshieh** [comment]
merge it as it will unblock whisper CI (170 -> 20 failures) + (quite) some pipeline tests

**Pull request merged**